javascript-es2015/src/chapter1/ch1-q6.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+/**
+ * Takes an input string and counts contiguous sequences of the same character
+ * and replaces them with XC (X = count, C = character).
+ *
+ * N = |str|
+ * Time: O(N)
+ * Additional space: O(N)
+ *
+ * @param  {string} str [description]
+ * @return {[type]}     [description]
+ */
+export function compressString(str) {
+  if (!str) {
+    return str;
+  }
+
+  let cStr = '';
+  for (let i = 0; i < str.length; ++i) {
+    let char = str[i],
+      start = i;
+    while (i + 1 < str.length && char === str[i + 1]) {
+      ++i;
+    }
+    // JS does not have a StringBuilder/StringBuffer style class for creating strings
+    // string concatenation has been heavily optimised in JS implementations and
+    // is faster than creating a string via an array then using a .join('') at the end
+    cStr += (i - start + 1) + char;
```

Comment:
Just a detail, in order to get 'a2' instead of '2a', it should be:
cStr += char + (i - start + 1);

Thx for this crazy huge work, solving those algo in ES6 is a great exercice!